netmempool/mempool.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 lesismal. All rights reserved.
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file.
+
+package netmempool
+
+import (
+	"github.com/lesismal/nbio/mempool"
+)
+
+type Buffer struct {
+	data        []byte
+	offset      int
+	notFromPool bool
+}
+
+func WrapBuffer(buf []byte) *Buffer {
+	return &Buffer{
+		data:        buf,
+		notFromPool: true,
+	}
+}
+
+func (b *Buffer) Remaining() []byte {
+	return b.data[b.offset:]
+}
+func (b *Buffer) Consumed(i int) {
+	b.offset += i
+}
+
+// Malloc borrows []byte from pool
+func Malloc(size int) *Buffer {
+	buf := mempool.Malloc(size)
+	return &Buffer{
```

Comment:
add
```golang
buffer := bufferPool.Get().(*Buffer)
```
and 
```golang
bufferPool.Put(buffer)
```
would be better

Reply:
do you mean instead of Malloc and Free. IE
rename Malloc->Get and Free->Put

Reply:
> do you mean instead of Malloc and Free. IE
> rename Malloc->Get and Free->Put

not that, I mean *Buffer objects should be managed by a pool either.